synthesizer/process/src/stack/registers/load.rs

```diff
@@ -126,6 +142,26 @@ impl<N: Network, A: circuit::Aleo<Network = N>> RegistersLoadCircuit<N, A> for R
             Operand::BlockHeight => bail!("Cannot load the block height in a non-finalize context"),
             // If the operand is the network ID, throw an error.
             Operand::NetworkID => bail!("Cannot load the network ID in a non-finalize context"),
+            // If the operand is the checksum, load the value of the checksum.
+            Operand::Checksum(program_id) => {
```

Comment:
Are we using checksum and edition in a non-finalize context? Consider bailing like we do above.

Reply:
Yes, we use in both on-chain and off-chain contexts.